lib/yocaml/key_value.ml

```diff
@@ -174,11 +182,14 @@ module Make_validator (KV : VALIDABLE) = struct
   let integer = integer_and Validate.valid
   let float = float_and Validate.valid
   let text = text_and Validate.valid
+  let null = null_and Validate.valid
 
   let optional_aux kind case_sensitive validator key subject =
     Option.fold
       ~none:(Validate.valid None)
-      ~some:validator
+      ~some:
+        Validate.Alt.(
+          fun x -> null_and (fun () -> Validate.valid None) x <|> validator x)
       (find_assoc ~case_sensitive key subject)
     |> hook_on_invalid_field kind key
```

Comment:
Could you add a test that validates that Null is taken into account when searching for an optional field?